addon/components/o-s-s/side-panel.ts

```diff
@@ -0,0 +1,54 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { run } from '@ember/runloop';
+
+interface OssSidePanelArgs {
+  visible: boolean;
+  backdrop: boolean;
+  size?: 'sm' | 'md';
+  onClose(): void;
+}
+
+export default class OssSidePanel extends Component<OssSidePanelArgs> {
+  @tracked showReturnButton: boolean = false;
+  declare initialTarget: HTMLElement | null;
+
```

Comment:
We should also ensure the onClose function is passed to the component here, as we do with the modal component for instance.

```
  constructor(owner: unknown, args: OssSidePanelArgs) {
    super(owner, args);

    assert('[component][OSS::SIdePanel] The close function is mandatory', args.close);
  }
```